ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/P2TargetUtils.java

```diff
@@ -1008,14 +1008,17 @@ static IQueryable<IInstallableUnit> getQueryableMetadata(URI[] repos, IProgressM
 		}
 
 		int repoCount = repos.length;
-		SubMonitor subMonitor = SubMonitor.convert(monitor, repoCount);
-
+		SubMonitor subMonitor = SubMonitor.convert(monitor, repoCount * 2);
 
+		Set<IRepositoryReference> seen = new HashSet<>();
 		List<IMetadataRepository> result = new ArrayList<>(repoCount);
+		List<IMetadataRepository> additional = new ArrayList<>();
```

Comment:
Can you tell the reason why the references are collected in a separate list? To always add them last?

Reply:
The reason is that the code explicitly checks that all given URLs could be loaded (some lines below), this could maybe archived smarter but having to list do not hurt much...